src/evict/evict_page.c

```diff
@@ -713,17 +727,6 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
 
     WT_RET(ret);
 
-    /*
-     * Give up on eviction during a checkpoint if the page splits.
-     *
-     * We get here if checkpoint reads a page with history store entries: if more of those entries
-     * are visible now than when the original eviction happened, the page could split. In most
-     * workloads, this is very unlikely. However, since checkpoint is partway through reconciling
-     * the parent page, a split can corrupt the checkpoint.
-     */
-    if (WT_SESSION_BTREE_SYNC(session) && page->modify->rec_result == WT_PM_REC_MULTIBLOCK)
-        return (__wt_set_return(session, EBUSY));
-
```

Comment:
I think this change can go, because it was a weird case before where checkpoint would read a page back in to apply lookaside changes, then find that the page split.  That doesn't happen any more.

If some checks are needed, I don't think this is the right place for them because `__evict_review` returns early for pages that don't need reconciliation.  That means a page left in cache after one reconciliation could skip this check the next time through and go on to split unexpectedly.